src/response/trowtoken.rs

```diff
@@ -0,0 +1,100 @@
+use failure::{Error};
+use std::io::Cursor;
+use std::time::{SystemTime, UNIX_EPOCH};
+use rocket::http::{Status};
+use rocket::http::{ContentType};
+use rocket::request::Request;
+use rocket::response::{Responder, Response};
+use crypto::sha2::Sha256;
+use chrono::Local;
+use jwt::{Header as TokenHeader, Token};
+
+const AUTHORISATION_SECRET: &str = "Bob Marley Rastafaria";
+
+#[derive(Debug, Serialize)]
+pub struct TrowToken;
+
+#[derive(Debug, Serialize, RustcEncodable, RustcDecodable)]
+pub struct BearerToken {
+    user_id: String,
+    client_id: String,
+    scope: String,
+    iat: u64,
+    exp: u64,
+}
+
+fn encode_token() -> Result<String, Error> {
+    // hard coded values for token
+    let username = "admin";
+    let client_id = "docker";
+    let scope = "push/pull";
+    let now = SystemTime::now();
+    let current_time = now.duration_since(UNIX_EPOCH).expect("Time went backwards");
+    let expiry_time = 3600;
```

Comment:
Can we add a comment with what this is in seconds?

Reply:
At the moment it is just a meaningless number! The issue time and expiry time aren't implemented. I ended up having to use the chrono crate to create a date formatted string that Docker would understand. At this stage, I just wanted to create a token and pass it back to Docker, then have Docker connect with the token and to verify it on trow. To do more advanced token checking, I may have to switch token crates as the main one seems to be very basic.